pkg/controller/mysql/database/reconciler.go

```diff
@@ -149,8 +150,20 @@ func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalCreation{}, errors.New(errNotDatabase)
 	}
 
-	err := c.db.Exec(ctx, xsql.Query{String: "CREATE DATABASE " + mysql.QuoteIdentifier(meta.GetExternalName(cr))})
-	return managed.ExternalCreation{}, errors.Wrap(err, errCreateDB)
+	binlog := defaultBinlog(cr.Spec.ForProvider.BinLog)
+	if !binlog {
+		if err := c.db.Exec(ctx, xsql.Query{
+			String: "SET sql_log_bin = 0",
+		}); err != nil {
+			return managed.ExternalCreation{}, errors.Wrap(err, errSetSQLLogBin)
+		}
+	}
```

Comment:
It seems this code is repeated a few times, can it be abstracted to `pkg/clients/mysql` perhaps?

Reply:
Great idea! The only downside I see is that `func (c mySQLDB) Exec(ctx context.Context, q xsql.Query) error` would have to change is signature in order to access `cr.Spec.ForProvider.BinLog` boolean value. Is that refactor acceptable?

Reply:
Since `func (c mySQLDB) Exec(ctx context.Context, q xsql.Query) error` is part of `xsql.DB` interface, which is common to mysql, mssql and postgres clients it would be very costly to refactor its signature. As every mock, call and implementation of that method should be updated. 

Consequently, I preferred another refactoring alternative to make it a bit less repetitive, creating a new function in `pkg/clients/mysql` called `ExecWithBinlogAndFlush` that is a wrapper function for `xsql.DB.Exec()` that allows disabling binlogs before the query and flushing privileges after the query (both optionally, as they default to true). 

What do you think? Would you consider this approach better than the previously mentioned refactor? I'm open to any further suggestions

Implemented in https://github.com/crossplane-contrib/provider-sql/pull/135/commits/0ba831f8ef0f2ec78b17f29b9957113bc11da51e